From 03ead57f7f2d6f5630fdff520f8e5970c18945de Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Tue, 9 Apr 2024 22:04:57 +0800
Subject: [PATCH 4/5] fix(core/launcher.cpp): revert to old HiDPI handling
 behaviour

The current implementation makes icons blurry in Qt 5 builds, whilst
fractional scaling is enabled.

Co-authored-by: Henry Chen <chenx97@aosc.io>
---
 Telegram/SourceFiles/core/launcher.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Telegram/SourceFiles/core/launcher.cpp b/Telegram/SourceFiles/core/launcher.cpp
index 2765fe3..5aaafb0 100644
--- a/Telegram/SourceFiles/core/launcher.cpp
+++ b/Telegram/SourceFiles/core/launcher.cpp
@@ -338,7 +338,9 @@ void Launcher::initHighDpi() {
 #endif // Qt < 6.2.0
 
 #if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
-	QApplication::setAttribute(Qt::AA_EnableHighDpiScaling, true);
+	QApplication::setAttribute(Qt::AA_DisableHighDpiScaling, true);
+	QApplication::setHighDpiScaleFactorRoundingPolicy(
+		Qt::HighDpiScaleFactorRoundingPolicy::Floor);
 #endif // Qt < 6.0.0
 
 	if (OptionFractionalScalingEnabled.value()) {
-- 
2.44.0

